File: PabellonFrontEnd/src/app/models/Product.ts
import { Options } from "./Options";

export class Product {
    id: number = 0;
    name: string = '';
    image: string = '';
    price: number = 0;
    catalogId: string = '';
    options: Options[] = [];
    description: string = '';
    disabled: boolean = false;
    quantity: number = 0
}

export class CustomizedProduct {
    id: string = generateUUID();
    product: Product;
    selectedOptions: Options[] = [];
    selectedQuantities: { [optionId: number]: number } = {};

    constructor(product: Product) {
        this.product = product;
    }

    getTotalPrice(): number {
        let total = this.product.price;
        this.selectedOptions.forEach(opt => total += opt.price);
        return total;
    }
}

function generateUUID(): string {
  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
    const r = Math.random() * 16 | 0, v = c === 'x' ? r : (r & 0x3 | 0x8);
    return v.toString(16);
  });
}

File: PabellonWebAPI/Pabellon.Context.Core/Models/Product.cs
using Pabellon.Context.Core.Models;

namespace Pabellon.Core.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Image {  get; set; }
        public double Price { get; set; }
        public string CatalogId { get; set; }     
        public string? Description { get; set; }
        public bool Disabled { get; set; }

        public Catalog Catalog { get; set; }
        public ICollection<Option> Options { get; set; } = new List<Option>();
    }
}
